File: Unit2-01/java/MrCoxallStack.java
/*
 * This class defines a string stack
 *
 * @author  Troy Appleby
 * @version 1.0
 * @since   2024-04-07
 */

import java.util.ArrayList;
import java.util.List;

/**
 * This is the class.
*/
public class MrCoxallStack {
    /**
     * Adds strings to an array.
     *
     * @param array contains the array of strings
     */
    private List<String> stackAsList;

    /**
    * Constructs a new instance of MrCoxallStack.
    * Initializes the stack with an empty ArrayList.
    *
    */
    public MrCoxallStack() {
        stackAsList = new ArrayList<>();
    }

    /**
     * The push method.
     *
     * @param input from Main.java
    */
    public void push(String input) {
        stackAsList.add(input);
    }

    /**
     * The showStack method.
     *
     * @return the values as a string
     */
    public String showStack() {
        String values = "";
        for (int counter = 0; counter < stackAsList.size(); counter++) {
            values += stackAsList.get(counter);
            if (counter < stackAsList.size() - 1) {
                values += ", ";
            }

        return values;
    }
}

File: Unit2-01/MrCoxallStack.ts
/**
 * The class defines a string stack
 *
 * author  Troy Appleby
 * version 1.0
 * since   2024-03-22
 */

export class MrCoxallStack {
  // strsstack: string[]

  // variables
  constructor() {
    this.stackAsList = []
  }

  // push item to end of list
  push(input) {
    this.stackAsList.push(input)
  }

  // show full stack seperated by commas
  showStack() {
    let values = ""
    for (let counter = 0; counter < this.stackAsList.length; counter++) {
      values = values + this.stackAsList[counter] + ", "
    }
    values = values.substring(0, values.length-2)
    return values
  }
}
